tests/test_strong_branching.py

```diff
@@ -0,0 +1,201 @@
+                if not downinf and downvalid:
+                    self.scip.updateVarPseudoCost(branch_cands[i], -self.scip.frac(lpsol), downgain, 1)
+                if not upinf and upvalid:
+                    self.scip.updateVarPseudoCost(branch_cands[i], 1 - self.scip.frac(lpsol), upgain, 1)
+
+            scores[i] = self.scip.getBranchScoreMultiple(branch_cands[i], [downgain, upgain])
+            if scores[i] > scores[best_cand_idx]:
+                best_cand_idx = i
+
+        # End strong branching
+        self.scip.endStrongBranching()
+
+        # In the case of an LP error
+        if lperror:
+            return {"result": SCIP_RESULT.DIDNOTRUN}
+
+        # Branch on the variable with the largest score
+        down_child, eq_child, up_child = self.model.branchVarVal(
+            branch_cands[best_cand_idx], branch_cands[best_cand_idx].getLPSol())
+
+        # Update the bounds of the down node and up node
+        if self.scip.allColsInLP() and not self.idempotent:
+            if down_child is not None and down_bounds[best_cand_idx] is not None:
+                self.scip.updateNodeLowerbound(down_child, down_bounds[best_cand_idx])
+            if up_child is not None and up_bounds[best_cand_idx] is not None:
+                self.scip.updateNodeLowerbound(up_child, up_bounds[best_cand_idx])
+
+        return {"result": SCIP_RESULT.BRANCHED}
+
+
+class FeatureSelectorBranchingRule(Branchrule):
+
+    def __init__(self, scip):
+        self.scip = scip
+
+    def branchexeclp(self, allowaddcons):
+
+        if self.scip.getNNodes() == 1 or self.scip.getNNodes() == 250:
+
+            rows = self.scip.getLPRowsData()
+            cols = self.scip.getLPColsData()
+
+            # This is just a dummy rule to check functionality.
+            # A user should be able to see how they can access information without affecting the solve process.
+
+            age_row = rows[0].getAge()
+            age_col = cols[0].getAge()
+            red_cost_col = self.scip.getColRedCost(cols[0])
+
+            avg_sol = cols[0].getVar().getAvgSol()
+
+        return {"result": SCIP_RESULT.DIDNOTRUN}
+
+
+def create_model():
+    scip = Model()
+    scip.setHeuristics(SCIP_PARAMSETTING.OFF)
+    scip.setSeparating(SCIP_PARAMSETTING.OFF)
+    scip.setLongintParam("limits/nodes", 2000)
+
+    x0 = scip.addVar(lb=-2, ub=4)
+    r1 = scip.addVar()
+    r2 = scip.addVar()
+    y0 = scip.addVar(lb=3)
+    t = scip.addVar(lb=None)
+    l = scip.addVar(vtype="I", lb=-9, ub=18)
+    u = scip.addVar(vtype="I", lb=-3, ub=99)
+
+    more_vars = []
+    for i in range(100):
+        more_vars.append(scip.addVar(vtype="I", lb=-12, ub=40))
+        scip.addCons(quicksum(v for v in more_vars) <= (40 - i) * quicksum(v for v in more_vars[::2]))
+
+    for i in range(100):
+        more_vars.append(scip.addVar(vtype="I", lb=-52, ub=10))
+        scip.addCons(quicksum(v for v in more_vars[50::2]) <= (40 - i) * quicksum(v for v in more_vars[405::2]))
```

Comment:
Possible typo on ```more_vars[405::2]```. There are only 200 ```more_vars``` right?

Reply:
This was an oversight by me..... I wonder why it didn't raise an error......
I copied it from `test_branch_probing` and just lowered the number of constraints to make the instance simpler.

Reply:
Changed to 200